include/curl/typecheck-gcc.h

```diff
@@ -367,8 +377,16 @@ _CURL_WARNING(_curl_easy_getinfo_err_curl_socket,
   (CURLINFO_DOUBLE < (info) && (info) < CURLINFO_SLIST)
 
 /* true if info expects a pointer to struct curl_slist * argument */
-#define _curl_is_slist_info(info)                                             \
-  (CURLINFO_SLIST < (info) && (info) < CURLINFO_SOCKET)
+#define _curl_is_slist_info(info)                                       \
+  (CURLINFO_SLIST < (info) && (info) < CURLINFO_SOCKET &&               \
+   ((info & CURLINFO_MASK) < (CURLINFO_CERTINFO & CURLINFO_MASK)))
```

Comment:
Yes I think we should switch the approach for the SLIST ones and consider them to be generic pointers and change the slist check there to check for the specific info arguments explicitly instead. I'll do that and also clarify that with a comment.